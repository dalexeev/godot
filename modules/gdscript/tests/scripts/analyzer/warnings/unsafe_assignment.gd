func test():
	var _hard_int: int = 1
	var _variant_int: Variant = 1
	var _weak_int = 1

	_hard_int = _hard_int
	_hard_int = _variant_int
	_hard_int = _weak_int

	_variant_int = _hard_int
	_variant_int = _variant_int
	_variant_int = _weak_int

	_weak_int = _hard_int
	_weak_int = _variant_int
	_weak_int = _weak_int

	_hard_int += _hard_int
	_hard_int += _variant_int
	_hard_int += _weak_int

	_variant_int += _hard_int
	_variant_int += _variant_int
	_variant_int += _weak_int

	_weak_int += _hard_int
	_weak_int += _variant_int
	_weak_int += _weak_int
