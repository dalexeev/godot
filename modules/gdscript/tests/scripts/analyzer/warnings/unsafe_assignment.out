GDTEST_OK

>> WARNING
>> Line: 7
>> UNSAFE_ASSIGNMENT
>> The left operand is of the subtype "int" but a value of the supertype "Variant" was assigned.

>> WARNING
>> Line: 8
>> UNSAFE_ASSIGNMENT
>> The left operand is of the subtype "int" but a value of the supertype "Variant" was assigned.

